bridges/snowbridge/parachain/pallets/outbound-queue/src/lib.rs

```diff
@@ -0,0 +1,413 @@
+//! Pallet for committing outbound messages for delivery to Ethereum
+//!
+//! # Overview
+//!
+//! Messages come either from sibling parachains via XCM, or BridgeHub itself
+//! via the `snowbridge-system` pallet:
+//!
+//! 1. `snowbridge_router_primitives::outbound::EthereumBlobExporter::deliver`
+//! 2. `snowbridge_system::Pallet::send`
+//!
+//! The message submission pipeline works like this:
+//! 1. The message is first validated via the implementation for
+//!    [`snowbridge_core::outbound::SendMessage::validate`]
+//! 2. The message is then enqueued for later processing via the implementation for
+//!    [`snowbridge_core::outbound::SendMessage::deliver`]
+//! 3. The underlying message queue is implemented by [`Config::MessageQueue`]
+//! 4. The message queue delivers messages back to this pallet via the implementation for
+//!    [`frame_support::traits::ProcessMessage::process_message`]
+//! 5. The message is processed in `Pallet::do_process_message`: a. Assigned a nonce b. ABI-encoded,
+//!    hashed, and stored in the `MessageLeaves` vector
+//! 6. At the end of the block, a merkle root is constructed from all the leaves in `MessageLeaves`.
+//! 7. This merkle root is inserted into the parachain header as a digest item
+//! 8. Offchain relayers are able to relay the message to Ethereum after: a. Generating a merkle
+//!    proof for the committed message using the `prove_message` runtime API b. Reading the actual
+//!    message content from the `Messages` vector in storage
+//!
+//! On the Ethereum side, the message root is ultimately the thing being
+//! verified by the Polkadot light client.
+//!
+//! # Message Priorities
+//!
+//! The processing of governance commands can never be halted. This effectively
+//! allows us to pause processing of normal user messages while still allowing
+//! governance commands to be sent to Ethereum.
+//!
+//! # Fees
+//!
+//! An upfront fee must be paid for delivering a message. This fee covers several
+//! components:
+//! 1. The weight of processing the message locally
+//! 2. The gas refund paid out to relayers for message submission
+//! 3. An additional reward paid out to relayers for message submission
+//!
+//! Messages are weighed to determine the maximum amount of gas they could
+//! consume on Ethereum. Using this upper bound, a final fee can be calculated.
+//!
+//! The fee calculation also requires the following parameters:
+//! * ETH/DOT exchange rate
+//! * Ether fee per unit of gas
+//!
+//! By design, it is expected that governance should manually update these
+//! parameters every few weeks using the `set_fee_parameters` extrinsic in the
+//! system pallet.
+//!
+//! ## Fee Computation Function
+//!
+//! ```text
+//! LocalFee(Message) = WeightToFee(ProcessMessageWeight(Message))
+//! RemoteFee(Message) = MaxGasRequired(Message) * FeePerGas + Reward
+//! Fee(Message) = LocalFee(Message) + (RemoteFee(Message) / Ratio("ETH/DOT"))
+//! ```
```

Comment:
Where do you test claiming / payout of the relayer rewards?

Reply:
For the Polkadot->Ethereum direction (outbound-queue), the reward is done on the Ethereum side and is tested in the solidity code. [[test case](https://github.com/Snowfork/snowbridge/blob/main/contracts/test/Gateway.t.sol#L252-L253)]

For the Ethereum->Polkadot direction (inbound-queue), the reward is done on the Polkadot side. I have added a test case to check this in [PR #1081](https://github.com/Snowfork/snowbridge/pull/1081).

Reply:
Thank you for making the inbound-queue reward test strict (https://github.com/Snowfork/snowbridge/pull/1081#discussion_r1433989885, https://github.com/Snowfork/snowbridge/pull/1086).

Can you do the same for the outbound direction, which is currently just `>`/`<`?